packages/assets-controllers/src/token-prices-service/codefi-v2.ts

```diff
@@ -474,64 +492,68 @@ export class CodefiTokenPricesServiceV2
    * given addresses which are expected to live on the given chain.
    *
    * @param args - The arguments to function.
-   * @param args.chainId - An EIP-155 chain ID.
-   * @param args.tokenAddresses - Addresses for tokens that live on the chain.
+   * @param args.assets - The assets to get prices for.
    * @param args.currency - The desired currency of the token prices.
    * @returns The prices for the requested tokens.
    */
   async fetchTokenPrices({
-    chainId,
-    tokenAddresses,
+    assets,
     currency,
   }: {
-    chainId: SupportedChainId;
-    tokenAddresses: Hex[];
+    assets: EvmAssetAddressWithChain<SupportedChainId>[];
     currency: SupportedCurrency;
-  }): Promise<Partial<TokenPricesByTokenAddress<Hex, SupportedCurrency>>> {
-    const chainIdAsNumber = hexToNumber(chainId);
+  }): Promise<EvmAssetWithMarketData<SupportedChainId, SupportedCurrency>[]> {
+    const assetsWithIds: EvmAssetWithId<SupportedChainId>[] = assets.map(
+      (asset) => {
+        const caipChainId = toCaipChainId(
+          KnownCaipNamespace.Eip155,
+          hexToNumber(asset.chainId).toString(),
+        );
+
+        const nativeAddress = getNativeTokenAddress(asset.chainId);
 
-    const url = new URL(`${BASE_URL}/chains/${chainIdAsNumber}/spot-prices`);
+        return {
+          ...asset,
+          assetId:
+            nativeAddress.toLowerCase() === asset.tokenAddress.toLowerCase()
+              ? HEX_CHAIN_ID_TO_CAIP19_NATIVE_ASSET_MAP[asset.chainId]
+              : `${caipChainId}/erc20:${asset.tokenAddress.toLowerCase()}`,
+        };
+      },
+    );
+
+    const url = new URL(`${BASE_URL_V3}/spot-prices`);
     url.searchParams.append(
-      'tokenAddresses',
-      [getNativeTokenAddress(chainId), ...tokenAddresses].join(','),
```

Comment:
The service no longer gets a chain or list of chains, so it is not responsible of also adding native assets, those assets needs to be passed as part of the list.